compiler/noirc_frontend/src/tests.rs

```diff
@@ -213,12 +241,31 @@ fn check_errors_with_options(src: &str, allow_parser_errors: bool, options: Fron
         secondary_spans_with_errors.into_iter().collect();
 
     let src = code_lines.join("\n");
-    let (_, _, errors) = get_program_with_options(&src, allow_parser_errors, options);
+    let (_, mut context, errors) = get_program_with_options(&src, allow_parser_errors, options);
+    let mut errors = errors.iter().map(CustomDiagnostic::from).collect::<Vec<_>>();
+
+    if monomorphize {
+        if !errors.is_empty() {
+            panic!("Expected no errors before monomorphization, got: {:?}", errors);
```

Comment:
`Debug`-printed errors are hard to read: is there an easy way to pretty-print errors here?